pandas/core/indexes/timedeltas.py

```diff
@@ -166,6 +167,19 @@ def __new__(cls, data=None, unit=None, freq=None, start=None, end=None,
         elif copy:
             data = np.array(data, copy=True)
 
+        data = np.array(data, copy=False)
+        if data.dtype == np.object_:
+            data = array_to_timedelta64(data)
```

Comment:
why are these checks NOT done in _simple_new? this is inconsistent with other code.

We should be really really clear on what is acceptable in _simple_new vs. what is not. IIRC from another of your PR's you did checks on object type in _simple_new for example.

Reply:
AFAICT the current verbose-checking is largely driven by the weird cases (that these PRs get rid of) where `None` or `[]` is passed to `_shallow_copy`.

This and the associated DatetimeIndex PR impose a simple/strict API for _simple_new: it expects an np.ndarray that may be either i8 or M8[ns]/m8[ns].